### PR TITLE
fix(nextcloud): create file even if webdav response is empty

### PR DESCRIPTION
**Pull request by Leptopoda**
fixes: #1472

### CODE REVIEW

File: packages/nextcloud/lib/src/webdav/client.dart [hunk 1]
```diff
@@ -232,29 +232,36 @@ class WebDavClient {
...
-        }
-      });
-      await completer.future;
-      await sink.close();
+
+    if (contentLength == null || contentLength <= 0) {
+      await file.create();
+      onProgress?.call(1);
+      return;
     }
```
Comment (provokateurin) [inline packages/nextcloud/lib/src/webdav/client.dart:249]: Did you test if just opening the sink, adding no elements and then closing it also creates the file? Then we wouldn't need an special handling

Comment (Leptopoda) [reply packages/nextcloud/lib/src/webdav/client.dart:249]: Yep this is working but we would still need to emit the `1` as the progress again creating a separate condition:
Is this really easier to read?

```
  Future<void> getFile(
    PathUri path,
    File file, {
    void Function(double progress)? onProgress,
  }) async {
    final response = await getStream(path);
    final contentLength = response.contentLength;

    final sink = file.openWrite();

    if (contentLength != null && contentLength > 0) {
      final completer = Completer<void>();
      var downloaded = 0;

      response.stream.listen((chunk) async {
        sink.add(chunk);
        downloaded += chunk.length;
        onProgress?.call(downloaded / contentLength);
        if (downloaded >= contentLength) {
          completer.complete();
        }
      });

      await completer.future;
    } else {
      onProgress?.call(1);
    }

    await sink.close();
  }
```

File: packages/nextcloud/test/webdav_test.dart [hunk 2]
```diff
@@ -653,6 +653,26 @@ void main() {
...
+
+        await client.webdav.putFile(
+          source,
+          source.statSync(),
+          PathUri.parse('empty-file'),
+        );
+        await client.webdav.getFile(
+          PathUri.parse('empty-file'),
+          destination,
+          onProgress: (progress) => expect(progress, equals(1)),
```
Comment (Leptopoda) [inline packages/nextcloud/test/webdav_test.dart:670]: This test case needs to be changed as it only checks that the progress equals `1` if called and does not check that the progress is called at all.

File: packages/nextcloud/lib/src/webdav/client.dart (hunk 1 above)
Comment (provokateurin) [reply packages/nextcloud/lib/src/webdav/client.dart:249]: I forgot about that, then this is indeed cleaner.

**Review by provokateurin — APPROVED** [review]

**Pull request merged**